CodeEditModules/Modules/WorkspaceClient/src/Model/FileItem.swift

```diff
@@ -257,7 +257,13 @@ public extension WorkspaceClient {
             var fileUrl = self.url
             while FileItem.fileManger.fileExists(atPath: fileUrl.path) {
                 let previousName = fileUrl.lastPathComponent
-                fileUrl = fileUrl.deletingLastPathComponent().appendingPathComponent("\(previousName) copy")
+                let previousNameWithoutExtension = (previousName as NSString).deletingPathExtension
+                let filextension = fileUrl.pathExtension
+                let fileName = filextension.isEmpty ?
+                previousName : previousNameWithoutExtension
+                fileUrl = fileUrl.deletingLastPathComponent()
+                fileUrl.appendPathComponent("\(fileName)-copy")
+                fileUrl.appendPathExtension("\(filextension)")
```

Comment:
```suggestion
                fileUrl.appendPathExtension(filextension)
```

Reply:
Thanks for the spot. Late Friday changes got the better of me 😅